wasm.md

```diff
@@ -1531,6 +1510,238 @@ Start Function
+          ~> ( memory ID SPEC )
+          ...
+         </k>
+```
+
+Imports
+-------
+
+Imports need to describe the type of what is imported.
+That an import is really a subtype of the declared import needs to be checked at instantiation time.
+The value of a global gets copied when it is imported.
+
+```k
+    syntax Defn       ::= ImportDefn
+    syntax ImportDefn ::= "(" "import" String String ImportDesc ")"
+    syntax ImportDesc ::= "(" "func"   OptionalId TypeUse        ")" [klabel(funcImportDesc)]
+                        | "(" "table"  OptionalId TableType      ")" [klabel( tabImportDesc)]
+                        | "(" "memory" OptionalId MemType        ")" [klabel( memImportDesc)]
+                        | "(" "global" OptionalId TextGlobalType ")" [klabel(globImportDesc)]
+ // -----------------------------------------------------------------------------------------
+    rule <k> ( import MOD NAME (func OID:OptionalId TUSE:TypeUse) ) => . ... </k>
+         <curModIdx> CUR </curModIdx>
+         <moduleInst>
+           <modIdx> CUR </modIdx>
+           <typeIds> TYPEIDS </typeIds>
+           <types>   TYPES   </types>
+           <funcIds> IDS => #saveId(IDS, OID, NEXT) </funcIds>
+           <funcIndices> FS => FS [NEXT <- ADDR] </funcIndices>
+           <nextFuncIdx> NEXT => NEXT +Int 1 </nextFuncIdx>
+           ...
+         </moduleInst>
+         <moduleRegistry> ... MOD |-> MODIDX ... </moduleRegistry>
+         <moduleInst>
+           <modIdx> MODIDX </modIdx>
+           <funcIds> IDS' </funcIds>
+           <funcIndices> ... #ContextLookup(IDS' , TFIDX) |-> ADDR ... </funcIndices>
+           <exports>     ... NAME |-> TFIDX                        ... </exports>
+           ...
+         </moduleInst>
+         <funcDef>
+           <fAddr> ADDR </fAddr>
+           <fType> FTYPE </fType>
+           ...
+         </funcDef>
+      requires unnameFuncType(FTYPE) ==K unnameFuncType(asFuncType(TYPEIDS, TYPES, TUSE))
+
+    rule <k> ( import MOD NAME (table OID:OptionalId (LIM _):TableType) ) => . ... </k>
+         <curModIdx> CUR </curModIdx>
+         <moduleInst>
+           <modIdx> CUR </modIdx>
+           <tabIds> IDS => #saveId(IDS, OID, 0) </tabIds>
+           <tabIndices> .Map => 0 |-> ADDR </tabIndices>
+           ...
+         </moduleInst>
+         <moduleRegistry> ... MOD |-> MODIDX ... </moduleRegistry>
+         <moduleInst>
+           <modIdx> MODIDX </modIdx>
+           <tabIds> IDS' </tabIds>
+           <tabIndices> ... #ContextLookup(IDS' , TFIDX) |-> ADDR ... </tabIndices>
+           <exports>    ... NAME |-> TFIDX                        ... </exports>
+           ...
+         </moduleInst>
+         <tabInst>
+           <tAddr> ADDR </tAddr>
+           <tmax>  MAX  </tmax>
+           <tsize> SIZE </tsize>
+           ...
+         </tabInst>
+       requires #limitsMatchImport(SIZE, MAX, LIM)
+
+    rule <k> ( import MOD NAME (memory OID:OptionalId (LIM:Limits):MemType) ) => . ... </k>
```

Comment:
What's up with `(LIM:Limits):MemType)`? AFAICT, `Limits` is the only subsort of `MemType`, so why do we have the intermediate sort `MemType`. But why are we sort checking it twice?

Reply:
Yeah, that's a mistake. Fixed.